cmd/node-cache/app/cache_app.go

```diff
@@ -269,6 +270,17 @@ func (c *CacheApp) setupNetworking() {
 }
 
 func (c *CacheApp) runPeriodic() {
+	// if a pidfile is defined in flags, setup iptables as soon as it's created
+	if c.params.Pidfile != "" {
+		if ok := waitForFile(c.params.Pidfile, time.Second*5, time.Second*1); !ok {
```

Comment:
If the pidfile is specified, we should block until the file is created. If it takes longer than 5s, we should still wait before starting the periodic loop, since coreDNS is still not running.

Can you also try this out on a test cluster with this image, to make sure the pidfile is indeed being created?

Reply:
I simplified the code a bit, and made the wait indefinite.
I also tested this on a cluster with and without the `pidfile` flag and it works as expected, I also exec'd into a container and verified the pidfile exist.
```
$ kubectl exec -it node-local-dns-2fhkm -- sh
# ls *.pid
coredns.pid
```